compliance/collection/nodeinventorizer/caching_inventorizer.go

```diff
@@ -51,8 +51,14 @@ func NewCachingScanner(analyzer NodeInventorizer, inventoryCachePath string, cac
 // A cached version is returned if it exists and is fresh enough.
 // Otherwise, a new scan guarded by a backoff is run by the injected analyzer.
 func (c *CachingScanner) Scan(nodeName string) (*storage.NodeInventory, error) {
-	// check whether a cached inventory exists that has not exceeded its validity
-	cachedInv, validUntil := readCachedInventory(c.inventoryCachePath)
+	// Check whether a cached inventory exists that has not exceeded its validity.
+	// On errors, we write a backoff file to ensure we wait before continuing.
+	cachedInv, validUntil, err := readCachedInventory(c.inventoryCachePath)
+	if err != nil {
+		if err := writeBackoff(c.maxBackoff, c.inventoryCachePath); err != nil {
+			return nil, errors.Wrap(err, "writing node scan backoff file")
+		}
+	}
```

Comment:
The error handling here complicates things quite a bit. I'm also not sure whether you want to write backoff in case the cache is corrupt/unreadable. It would be simpler to keep the code here as-is but deal with error inside of `readCachedInventory()`
```suggestion
	// Check whether a cached inventory exists that has not exceeded its validity.
	cachedInv, validUntil := readCachedInventory(c.inventoryCachePath)
```

Reply:
A bit more on this:

> I'm also not sure whether you want to write backoff in case the cache is corrupt/unreadable.

The current state of code is such that failed attempt to write cache _should_ leave backoff. Should != always will because software programs and hardware does not fail in binary way.

It is probably okay and desirable to _wait_ some backoff time in case the cache is corrupt.
From my reading of the current code, it most likely will happen because corrupt cache most probably means also error reading backoff which will then default the backoff value to the max backoff. One situation when that's not the case is when the cache **is** corrupt but backoff **is not** corrupt and is zero. How can that happen? I suppose only protobuf serialization/deserialization issues.

In such case you don't need to _write_ backoff file. Just assume the max value in case of `error` from reading cache.